File: task-03/sub-task04/Cpp/readPattern.cpp
#include <iostream>
#include <fstream>

int main() {
    std::string inputFilePath = "input.txt";
    std::string outputFilePath = "output.txt";
    int n;

    // Reading the number n from input.txt
    std::ifstream inputFile(inputFilePath);
    if (!inputFile) {
        std::cerr << "Error reading from input file." << std::endl;
        return 1;
    }
    inputFile >> n;
    inputFile.close();

    // Generating the diamond pattern and writing it to output.txt
    std::ofstream outputFile(outputFilePath);
    if (!outputFile) {
        std::cerr << "Error writing to output file." << std::endl;
        return 1;
    }

    // Upper part of the diamond
    for (int i = 1; i <= n; ++i) {
        outputFile << std::string(n - i, ' ') << std::string(2 * i - 1, '*') << "\n";
    }
    // Lower part of the diamond
    for (int i = n - 1; i >= 1; --i) {
        outputFile << std::string(n - i, ' ') << std::string(2 * i - 1, '*') << "\n";
    }

    outputFile.close();
    std::cout << "Diamond pattern generated and written to output.txt." << std::endl;
    return 0;
}

File: task-03/sub-task04/Go/readPattern.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	inputFilePath := "input.txt"
	outputFilePath := "output.txt"

	// Read the number n from input.txt
	file, err := os.Open(inputFilePath)
	if err != nil {
		fmt.Println("Error reading from input file.")
		return
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	scanner.Scan()
	n, err := strconv.Atoi(strings.TrimSpace(scanner.Text()))
	if err != nil {
		fmt.Println("Invalid number format.")
		return
	}

	// Generate the diamond pattern and write to output.txt
	outputFile, err := os.Create(outputFilePath)
	if err != nil {
		fmt.Println("Error writing to output file.")
		return
	}
	defer outputFile.Close()

	writer := bufio.NewWriter(outputFile)

	// Upper part of the diamond
	for i := 1; i <= n; i++ {
		fmt.Fprintln(writer, strings.Repeat(" ", n-i)+strings.Repeat("*", 2*i-1))
	}

	// Lower part of the diamond
	for i := n - 1; i >= 1; i-- {
		fmt.Fprintln(writer, strings.Repeat(" ", n-i)+strings.Repeat("*", 2*i-1))
	}

	writer.Flush()
	fmt.Println("Diamond pattern generated and written to output.txt.")
}

